lighthouse-core/audits/seo/canonical.js

```diff
@@ -10,6 +10,31 @@ const LinkHeader = require('http-link-header');
 const URL = require('../../lib/url-shim');
 const MainResource = require('../../computed/main-resource.js');
 const LINK_HEADER = 'link';
+const i18n = require('../../lib/i18n/i18n.js');
+
+const UIStrings = {
+  /** Title of a Lighthouse audit that provides detail on a page's rel=canonical link. This descriptive title is shown to users when the rel=canonical link is valid. */
+  title: 'Document has a valid `rel=canonical`',
+  /** Descriptive title of a Lighthouse audit that provides detail on a page's rel=canonical link. This imperative title is shown to users when the rel=canonical link is invalid and should be fixed. */
+  failureTitle: 'Document does not have a valid `rel=canonical`',
+  /** Description of a Lighthouse audit that tells the user *why* they need to have a valid rel=canonical link. This is displayed after a user expands the section to see more. No character length limits. 'Learn More' becomes link text to additional documentation. */
+  description: 'Canonical links suggest which URL to show in search results. ' +
+    '[Learn more](https://developers.google.com/web/tools/lighthouse/audits/canonical).',
+  /** [ICU Syntax] Explanatory message stating that there was a failure in an audit caused by multiple URLs conflicting with each other. */
+  explanationConflict: 'Multiple conflicting URLs ({urlList})',
+  /** [ICU Syntax] Explanatory message stating that there was a failure in an audit caused by a URL being invalid. */
+  explanationInvalid: 'Invalid URL ({url})',
+  /** [ICU Syntax] Explanatory message stating that there was a failure in an audit caused by a URL being relative instead of absolute. */
+  explanationRelative: 'Relative URL ({url})',
+  /** [ICU Syntax] Explanatory message stating that there was a failure in an audit caused by a URL pointing to a different hreflang than the current context. */
+  explanationPointsElsewhere: 'Points to another hreflang location ({href})',
+  /** [ICU Syntax] Explanatory message stating that there was a failure in an audit caused by a URL pointing to a different domain. */
+  explanationDifferentDomain: 'Points to a different domain ({url})',
+  /** Explanatory message stating that the page's canonical URL was pointing to a root of the same origin which is a common mistake. The canonical URL should point a different page with the same content whose URL is preferred, not a parent or root page. */
```

Comment:
this is confusing but my brain is tired from reviewing this PR now :) "root of the same origin" is the main thing that needs to be defined here (root directory and something about the origin...not sure if we have an existing origin we can reuse)